server/api/dev.js

```diff
@@ -0,0 +1,214 @@
+/*
+  Easier to refresh entire dev entry instead of upsert/update the dates
+  ??? What is the removed column for in entries?
+*/
+
+const db = require('../db');
+const { createLogger } = require('../logger');
+
+const NOW = new Date();
+const VOTE_MAX = 5;
+const BACKGROUND_COLORS = ['#000', '#FFF', '#4b91e3'];
+
+const logger = createLogger('API/DEV');
+
+const getDevContest = (status) => {
+  const contestStatuses = ['prerelease', 'reset', 'submission', 'review', 'voting', 'results'];
+  const contestIndex = contestStatuses.indexOf(status);
+  const isAfter = (lastStatus) => contestIndex > contestStatuses.indexOf(lastStatus);
+
+  const yearEnd = NOW.getMonth() === 11; // getMonth is zero-based
+  const date = NOW.toISOString().split('T')[0];
+  const fiveYears = 1000 * 60 * 60 * 24 * 365 * 5;
+  const pastest = new Date(NOW.getTime() - fiveYears - 1).toISOString();
+  const past = new Date(NOW.getTime() - fiveYears).toISOString();
+  const future = new Date(NOW.getTime() + fiveYears).toISOString();
+  const futurest = new Date(NOW.getTime() + fiveYears + 1).toISOString();
+
+  /*
+                s_start   s_end     v_start   v_end
+    prerelease  future    futurest  future    futurest
+    submission  pastest   futurest  future    futurest
+    review      pastest   past      future    futurest
+    voting      pastest   past      pastest   futurest
+    results     pastest   past      pastest   past
+  */
+
+  return {
+    id: 'dev',
+    name: status === 'reset' ? 'submission' : status,
+    date,
+    env_level: 'dev',
+    year_end: yearEnd,
+    winners_thread_id: null,
+    valid_reddit_id: false,
+    submission_start: isAfter('prerelease') ? pastest : future,
+    submission_end: isAfter('submission') ? past : futurest,
+    vote_start: (isAfter('review') ? pastest : future) + 2, // avoid vote before submission
+    vote_end: (isAfter('voting') ? past : futurest) + 2, // avoid vote before submission
+    subtext: 'Dev subtext',
+    local_voting: true,
+    prompt: 'This is the prompt for the development contest.',
+  };
+};
+
+exports.contest = async ({ body: { status }, username }, res) => {
+  try {
+    const isExisting = (await db.any("SELECT EXISTS (SELECT 1 FROM contests WHERE id='dev')"))[0].exists;
+
+    if (status === 'reset') {
+      /*
+        ------- CONTEST -------
+        Create a dev contest ONLY if it doesn't exit already
+        Reset should not change contest timestamps to avoid changing status
+      */
+      if (!isExisting) {
+        await db.insert('contests', [getDevContest(status)]);
+      }
+
+      /*
+        ------- DELETE OLD DATA -------
+        Ensures developer account is in dev database
+        Create a fake "dev" user
+      */
+      await db.del('votes', { contest_id: 'dev' });
+      await db.del('contest_entries', { contest_id: 'dev' });
+      await db.any("DELETE FROM entries WHERE id LIKE 'dev%';");
```

Comment:
Couldn't there possibly be real IDs that start with `'dev'`? There's no special characters in the UUID alphabet we're using, i.e. `[A-Za-z0-9]`, so maybe we should add a special character in the id like `dev-`.

Reply:
Good catch, used the hyphen as you suggested